doc/source/ray-air/examples/pytorch_resnet_batch_prediction.ipynb

```diff
@@ -0,0 +1,381 @@
+     ]
+    }
+   ],
+   "source": [
+    "predictions = batch_predictor.predict(\n",
+    "    ds,\n",
+    "    feature_columns=[\"image\"],\n",
+    "    keep_columns=[\"label\"],\n",
+    "    batch_size=128,\n",
+    "    max_scoring_workers=3,\n",
+    "    num_gpus_per_worker=1,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We specified several parameters in `predict()`:\n",
+    "- `feature_columns` specifies which columns are required for the model. The selected columns will be concatenated to build a batch tensor.\n",
+    "- The columns in `keep_columns` will be returned together with the prediction results. For example, you can keep image labels for evaluation later.\n",
+    "- The BatchPredictor uses CPUs for inference by default, please specify `num_gpus_per_worker` if you want to use GPUs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluating Prediction Accuracy\n",
+    "\n",
+    "`BatchPredictor.predict()` will return a Ray Dataset with a column of model output with key `\"predictions\"`, and all columns specified in `keep_columns`.\n",
+    "\n",
+    "In this example, the output of the ResNet model is a 1000-dimensional tensor containing the logits of each class. We'll measure accuracy with Top-1 and Top-5 accuracy.\n",
+    "(Top-N accuracy: The percentage of predictions where the true label falls in the top N predicted classes.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Map_Batches: 100%|██████████| 9/9 [00:00<00:00, 140.05it/s]\n",
+      "Shuffle Map: 100%|██████████| 9/9 [00:00<00:00, 667.80it/s]\n",
+      "Shuffle Reduce: 100%|██████████| 1/1 [00:00<00:00, 128.55it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Top-1 accuracy:  0.849\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Shuffle Map: 100%|██████████| 9/9 [00:00<00:00, 698.69it/s]\n",
+      "Shuffle Reduce: 100%|██████████| 1/1 [00:00<00:00, 126.10it/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Top-5 accuracy:  0.99\n"
+     ]
+    }
+   ],
+   "source": [
+    "def calculate_matches(batch: pd.DataFrame) -> pd.DataFrame:\n",
+    "    batch[\"top_5_pred\"] = batch[\"predictions\"].apply(lambda x: np.argsort(-x)[:5])\n",
+    "    batch[\"top_5_pred_name\"] = batch[\"top_5_pred\"].map(\n",
+    "        lambda x: [idx_to_class_name[idx] for idx in x]\n",
+    "    )\n",
+    "    batch[\"top_5_match\"] = batch.apply(lambda x: x[\"label\"] in x[\"top_5_pred\"], axis=1)\n",
+    "    batch[\"top_1_match\"] = batch.apply(\n",
+    "        lambda x: x[\"label\"] == x[\"top_5_pred\"][0], axis=1\n",
```

Comment:
is it slightly better to just do ``x["label"] == x["predictions"][0]`` instead?

Reply:
Actually `x["predictions"]` is a tensor of model output, which contains the logits for 1000 classes. We have to use `np.argsort` to pick the top1/top5 class id, then match with `x["label"]`.

Reply:
I see it now. 👌 
it's kinda hard to read code this way.